### PR TITLE
E2E tests for Allow lists with service invocation

### PR DESCRIPTION
**Pull request by vinayada1**
# Description

Added e2e tests for Allow lists with service invocation
Note: http to grpc test when tried via test infra was failing. I have verified this manually and will debug the test issue later.

#2075 

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2075 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by vinayada1** [comment]
Thanks @youngbupark for helping me with the setup for the manual verification of http to grpc test

File: tests/e2e/allowlists/allowlists_service_invocation_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	},
+	{
+		"Test deny with callee side http",
+		"allowlists-callee-http",
+		"opDeny",
+		"rpc error: code = PermissionDenied desc = access control policy has denied access to appid: allowlists-caller operation: opDeny verb: POST",
+	},
+}
+
+var moreServiceinvocationTests = []struct {
```
Comment (youngbupark) [inline tests/e2e/allowlists/allowlists_service_invocation_test.go:90]: I would rename `moreServiceInvocationTests` to `allowlistTests`

Comment (vinayada1) [reply tests/e2e/allowlists/allowlists_service_invocation_test.go]: fixed

**Review by youngbupark — APPROVED** [review]

**Pull request merged**